backend/tests/test_rocm_build.py

```
@@ -0,0 +1,126 @@
+            text=True,
+        )
+        assert result.returncode == 0
+        assert "--rocm" in result.stdout
+
+    def test_cannot_combine_cuda_and_rocm(self):
+        """Building with both CUDA and ROCm should raise ValueError."""
+        with pytest.raises(ValueError, match="Cannot build with both CUDA and ROCm"):
+            build_server(cuda=True, rocm=True)
+
+
+@pytest.mark.slow()
```

Ruff (0.15.10)

[warning] Use `@pytest.mark.slow` over `@pytest.mark.slow()`

Remove parentheses

(PT023)